docs/DeveloperGuide.md

```diff
@@ -1,34 +1,346 @@
+* *SubjectParser* - A class which reads the user commands to call the relevant Subject command.
+* *TaskParser* - A class which reads the user commands to call the relevant task command.
+* *TopicParser* - A class which reads the user commands to call the relevant topic command.
+* *card* - Package holds the necessary classes for the quiz, subject,topic and flashcard functionality.
+* *quiz* - Package holds the classes for the quiz functionality.
+* *Quiz* - An Abstract class that holds the result, list of flashcards and a list of incorrectAnswers for the 
+SubjectQuiz class and the TopicQuiz class to inherit from.
+* *SubjectQuiz* - Instantiated when the user calls for a quiz on a subject. 
+* *TopicQuiz* - Instantiated when the user calls for a quiz on a topic. 
+* *Result* - A class which stores the marks from a quiz and a description which indicates the performance of the user 
+for that quiz.
+* *ResultList* - A class which holds an ArrayList of results.
+* *Flashcard* - A class which instantiates the flashcard object.Contains the question and answer feature of the flashcard.
+* *Subject* - A class which instantiates the subject object. Holds the title of the subject and lists of flashcards, 
+results and topics.
+* *SubjectList* - A class which holds an ArrayList of Subjects.
+* *Topic* - A class which is instantiated when the user creates the topic. Holds the title of the topic, an arraylist of
+flashcards and an arraylist of results.
+* *TopicList* - A class which holds an ArrayList of Topics.
 
 ## Instructions for manual testing
 
-{Give instructions on how to do a manual product testing e.g., how to load sample data to be used for testing}
+### Testing the Jar File
+1. Refer to the [UserGuide](https://ay2021s1-cs2113t-w13-1.github.io/tp/UserGuide.html) on how to download the application.
+2. Open the application in command prompt.
+3. Try out various commands in the userguide to check if it works.
+4. Try out invalid commands and check if the application responds correspondingly.
+5. Exit the application and check the data files to check if all the data has been saved.
+6. Open the application again and check if the data has been loaded correctly. Use the `list` command for this step.
+
+### Testing the source code
+
+#### Setting up the project
+1. Download the source code from [here](https://github.com/AY2021S1-CS2113T-W13-1/tp/releases).
+2. The application is optimised for IntelliJ users. Download IntelliJ if you do not have it.
+3. Configure the Intellij for JDK 11, as described [here](https://se-education.org/guides/tutorials/intellijJdk.html).
+4. Import the project as a Gradle project, as described [here](https://se-education.org/guides/tutorials/intellijImportGradleProject.html).
+5. After the importing is complete, locate the `src/main/java/seedu.revised/Revised.java` file, right-click it, and choose
+`Run Revised.main()`.
+6. If the setup is correct, you should see soemthing like this:
```

Comment:
Should it be "something"? There are occasional typos and grammar mistakes, perhaps they could be fixed to improve readability?